Detectors/ITSMFT/MFT/base/include/MFTBase/GeometryTGeo.h

```diff
@@ -9,152 +9,178 @@
 // or submit itself to any jurisdiction.
 
 /// \file GeometryTGeo.h
-/// \brief A simple interface class to TGeoManager
-/// \author bogdan.vulpescu@cern.ch 
-/// \date 01/08/2016
+/// \brief Definition of the GeometryTGeo class
+/// \author bogdan.vulpescu@clermont.in2p3.fr - adapted from ITS, 21.09.2017
+
+#ifndef ALICEO2_MFT_GEOMETRYTGEO_H
+#define ALICEO2_MFT_GEOMETRYTGEO_H
+
+#include <vector>
+#include <array>
+#include <string>
+#include <TGeoMatrix.h> // for TGeoHMatrix
+#include <TObject.h>    // for TObject
+#include "DetectorsBase/DetID.h"
+#include "DetectorsBase/Utils.h"
+#include "DetectorsBase/GeometryManager.h"
+#include "ITSMFTBase/GeometryTGeo.h"
+#include "Rtypes.h" // for Int_t, Double_t, Bool_t, UInt_t, etc
+
+class TGeoPNEntry; 
+namespace o2 {
+namespace ITSMFT {
+  class Segmentation;
+  class SegmentationPixel;
+}
+}
 
-#ifndef ALICEO2_MFT_GEOMETRYTGEO_H_
-#define ALICEO2_MFT_GEOMETRYTGEO_H_
+namespace o2
+{
+  
+namespace MFT
+{
 
-#include <boost/bimap.hpp>
+class GeometryTGeo : public o2::ITSMFT::GeometryTGeo
+{
+ public:
 
-#include "TObject.h"
-#include "TString.h"
-#include "TObjArray.h"
-#include "TGeoMatrix.h"
+  typedef o2::Base::Transform3D Mat3D;
+  using DetMatrixCache::getMatrixT2L;
+  using DetMatrixCache::getMatrixL2G;
+  using DetMatrixCache::getMatrixT2G;
 
-namespace o2 {
-  namespace ITSMFT {
-    class Segmentation;
+  static GeometryTGeo* Instance() {
+    // get (create if needed) a unique instance of the object
+    if (!sInstance) sInstance = std::unique_ptr<GeometryTGeo>(new GeometryTGeo(true, true, 0));
+    return sInstance.get();
   }
-}
 
-namespace o2 {
-namespace MFT {
+  // adopt the unique instance from external raw pointer (to be used only to read saved instance from file)
+  static void adopt(GeometryTGeo* raw); 
 
-typedef boost::bimap<int,int> BimapType;
+  // constructor
+  // ATTENTION: this class is supposed to behave as a singleton, but to make it 
+  // root-persistent we must define public default constructor.
```

Comment:
do we need to make this class ROOT persistent?

Reply:
@sawenzel : this is copied from my implementation of similar ITS class. At the moment persistence is not used but I wanted to have it to be able initialize these classes w/o loading the geometry